Include yadda from npm instead of bower

sfbwalder opened the pull request:
This intent of this is to resolve issue #30.  At high level, the change is to use ember-browserify to allow yadda to be pulled in as an npm dependency rather than a bower dependency.  This eliminates  the use of the bower dependency that is distributed as a browserified module that overrides the global ```require()``` needed by ember.   The yadda npm package does change ember's ```require()```.

Summary of changes:
1. At install time, do not add yadda bower to the project, but rather add yadda npm package and ember-browserify add-on.
1. At install time add tests/helpers/yadda.js to the consuming application from the build print files.
1. Set the ember-browserify tests project to true so it looks at the test tree to browserify the yadda package.  This is merged with the consuming applications enviroment.js.
1. Add the yadda test helper in the dummy app so it can find the yadda package.

I did not update the readme.md, but will if this pull request is accepted.

ro0gr commented:
@sfbwalder have you tried something more lightweight than browserify? https://github.com/dfreeman/ember-cli-node-assets for example

Yadda has an umd distributive which we can just import with a shim I believe.

sfbwalder commented:
Yadda's umd distribution also overrides the global require [as shown here on 0.22.1](https://github.com/acuminous/yadda/blob/master/dist/yadda-umd-0.22.1.js#L1), and I'm not sure it's really a umd module.

I have not looked at ember-cli-node-assests, but will.

sfbwalder commented:
It looks like ember-cli-node-assets (and ember-cli-node-modules-to-vendor) are made to pull in a few files from an npm package, but not made to pull in an entire package built around commonjs modules as yadda is.

@ebryn thoughts?

ro0gr commented:
Sorry, I've missed the point about fighting with `require()`. 
Unfortunately node assets import is not a case here.

albertjan commented on tests/helpers/yadda.js:
```diff
@@ -0,0 +1,3 @@
+// needed for dummy app to run
```
do we not need this file in normal installations?

sfbwalder replied:
In a normal installation, the default blue print copies the equivalent file into the consuming apps tests/helpers folder.  It needed to be a blueprint file because browserify does not process the test-support tree.  And since the default blueprint is not run on the addon dummy app, this file needs to be there so the dummy app will work.

albertjan commented on index.js:
```diff
@@ -155,13 +155,5 @@ module.exports = {
   },
   included: function(app) {
     this._super.included(app);
-
```
this is the only change to the external api?

sfbwalder replied:
Correct.  The versioning can now be controlled by what ever version the consuming app has in the package.json.  There is no longer the need to pick a version from the browserified distribution of yadda.

albertjan commented:
@mschinis I'd like your input on this one 😄.

@sfbwalder How does the upgrade path look for people running the old version.

sfbwalder commented:
> How does the upgrade path look for people running the old version.

1. ```bower uninstall yadda --save```

1. ```ember install ember-cli-yadda@latest``` - also installs latest version of ember-browserify and yadda, and runs default blueprint.

1. If a specific version of yadda is being used:
    * Manually remove code from ember-cli-build.js that sets desired version
    * ```npm uninstall yadda --save-dev``` to remove latest version of yadda
    * ```npm install yadda@<desired version> --save-dev``` to get desired version

All existing features/steps work with no changes to upgrade.  Generating new features works just as it did, so no changes there either.

sfbwalder commented:
> @mschinis I'd like your input on this one :smile:.

albertjan commented:
Ehh if he doesn't respond I'll merge tomorrow 😄

mschinis commented:
egh. sorry guys. I missed this. I'll check it out tonight.

albertjan commented:
Hey @sfbwalder I'm gonna merge this, you ok with that?

sfbwalder commented:
@albertjan 

I just updated the readme, so you might check that out to see if it is satisfactory.

I also added tests/helpers/yadda.js to the mocha blueprint files.  We use qunit, not mocha, so I had overlooked that.

As far as testing, I have successfully used this change in an application that uses qunit, but not mocha.  I don't think it would make a difference which test framework is used.

I am good with merging it.

albertjan commented:
Thanks! 😄 I'll release a new version tomorrow. I'll also run a test with mocha.

albertjan commented:
@mschinis i felt this was moving forward, if you run into something let me know 😄

sfbwalder commented:
Thanks, I'll look for the new release!
